### PR TITLE
Add preconcurrency flag correctly on Linux

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

**Comment by adam-fowler** [comment]
@gwynne

**Comment by gwynne** [comment]
@adam-fowler IIRC, `@preconcurrency` ended up being broken on Swift 5.6, such that in NIO it's only used when `#if swift(>=5.7)` (ex.: https://github.com/apple/swift-nio/blob/main/Sources/NIOCore/EventLoopFuture.swift#L541-L601); is it deliberate that you're doing the exact opposite on non-Linux? 😕

**Comment by adam-fowler** [comment]
> @adam-fowler IIRC, `@preconcurrency` ended up being broken on Swift 5.6, such that in NIO it's only used when `#if swift(>=5.7)` (ex.: https://github.com/apple/swift-nio/blob/main/Sources/NIOCore/EventLoopFuture.swift#L541-L601); is it deliberate that you're doing the exact opposite on non-Linux? 😕

I thought `@preconcurrency` was fine for imports on 5.6. Anyway `Date` and `Data` were marked `Sendable` on macOS swift 5.7 and later which is why the check is the opposite way

File: Tests/SotoCoreTests/AWSRequestTests.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if compiler(<5.7)
+#if compiler(<5.7) || os(Linux)
```
Comment (gwynne) [inline Tests/SotoCoreTests/AWSRequestTests.swift:15]: Looks like you missed changing the `#if`s here and in `DictionaryEncoderTests.swift`, if this isn't on purpose

Comment (adam-fowler) [reply Tests/SotoCoreTests/AWSRequestTests.swift:15]: Yep looks like I missed that one

Comment (adam-fowler) [reply Tests/SotoCoreTests/AWSRequestTests.swift]: fixed

**Review by 0xTim — APPROVED** [review]

**Pull request merged**